src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -241,146 +291,220 @@ private void AddOptionIfAvailable<T>(Command command, OptionInfo<T> option)
-            false,
-            "Validate the program and options, but do not submit to Azure Quantum.");
+        /// <param name="entryPoint">The entry point.</param>
+        /// <returns>The OptionInfo for the --target option.</returns>
+        private OptionInfo<string?> CreateTargetOption(IEntryPoint entryPoint)
+        {
+            var targetAliases = ImmutableList.Create("--target");
+            const string targetDescription = "The target device ID.";
+            return string.IsNullOrWhiteSpace(entryPoint.DefaultExecutionTarget)
+                ? new OptionInfo<string?>(targetAliases, targetDescription)
+                : new OptionInfo<string?>(targetAliases, entryPoint.DefaultExecutionTarget, targetDescription);
+        }
 
         /// <summary>
-        /// The verbose option.
+        /// Simulates the entry point.
         /// </summary>
-        internal static readonly OptionInfo<bool> VerboseOption = new OptionInfo<bool>(
-            ImmutableList.Create("--verbose"), false, "Show additional information about the submission.");
+        /// <param name="parseResult">The command-line parsing result.</param>
+        /// <param name="simulator">The simulator to use.</param>
+        /// <param name="entryPoint">The entry point to simulate.</param>
+        /// <returns>The exit code.</returns>
+        private async Task<int> Simulate(ParseResult parseResult, string simulator, IEntryPoint entryPoint) =>
+            await entryPoint.Simulate(parseResult, settings, DefaultIfShadowed(entryPoint, this.CreateSimulatorOption(entryPoint), simulator));
 
         /// <summary>
-        /// Displays a message to the console using the given color and text writer.
+        /// Submits the entry point to Azure Quantum.
         /// </summary>
-        /// <param name="color">The text color.</param>
-        /// <param name="writer">The text writer for the console output stream.</param>
-        /// <param name="message">The message to display.</param>
-        internal static void DisplayWithColor(ConsoleColor color, TextWriter writer, string message)
-        {
-            var originalForeground = Console.ForegroundColor;
-            Console.ForegroundColor = color;
-            writer.WriteLine(message);
-            Console.ForegroundColor = originalForeground;
-        }
-    }
-
-    /// <summary>
-    /// A modification of the command-line <see cref="HelpBuilder"/> class.
-    /// </summary>
-    internal sealed class QsHelpBuilder : HelpBuilder
-    {
+        /// <param name="parseResult">The command-line parsing result.</param>
+        /// <param name="azureSettings">The Azure submission settings.</param>
+        /// <param name="entryPoint">The entry point to submit.</param>
+        /// <returns>The exit code.</returns>
+        private async Task<int> Submit(ParseResult parseResult, AzureSettings azureSettings, IEntryPoint entryPoint) =>
+            await entryPoint.Submit(parseResult, new AzureSettings
```

Comment:
nit: (I know this was in the original code, but) the `async` and `await` are unnecessary here.

Reply:
done